### PR TITLE
Length & speed based on units

### PR DESCRIPTION
**Pull request by kdiluca**
I merged from master but my only changes were in trace_attributes for units, length and speed.

### CODE REVIEW

File: src/thor/trace_attributes_action.cc [hunk 1]
```diff
@@ -21,12 +21,15 @@ using namespace valhalla::thor;
 
 
 namespace {
+  constexpr double kMilePerMeter = 0.000621371;
```
Comment (kevinkreiser) [inline src/thor/trace_attributes_action.cc]: these may be in midgard or baldr already

File: src/thor/trace_attributes_action.cc [hunk 2]
```diff
@@ -90,9 +93,16 @@ worker_t::result_t thor_worker_t::trace_attributes(
     trip_path = map_match();
   }
   json::MapPtr json;
+  auto id = request.get_optional<std::string>("id");
+  double distance_scale = kKmPerMeter;
+  double speed_scale = kKmPerHour;
+  auto units = request.get<std::string>("units", "km");
+  if (units == "mi") {
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: unless done somewhere else - will have to check for both "mi" or "miles"

Comment (kdiluca) [inline src/thor/trace_attributes_action.cc]: looking into this...

File: src/thor/trace_attributes_action.cc [hunk 3]
```diff
@@ -90,9 +92,15 @@ worker_t::result_t thor_worker_t::trace_attributes(
     trip_path = map_match();
   }
   json::MapPtr json;
+  auto id = request.get_optional<std::string>("id");
+  double distance_scale, speed_scale = 1;
+  auto units = request.get<std::string>("units", "km");
+  if (units == "mi") {
+    distance_scale = kMilePerMeter;
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: should be scale to go from km to miles - correct?

Comment (kdiluca) [inline src/thor/trace_attributes_action.cc]: k I believe I have it now and I'm using Midgards constant

File: src/thor/trace_attributes_action.cc [hunk 4]
```diff
@@ -90,9 +90,15 @@ worker_t::result_t thor_worker_t::trace_attributes(
     trip_path = map_match();
   }
   json::MapPtr json;
+  auto id = request.get_optional<std::string>("id");
+  //length defaults to km, speed defaults to km/h
+  double scale = 1;
+  auto units = request.get<std::string>("units", "km");
+  if (units == "mi") {
```
Comment (dgearhart) [inline src/thor/trace_attributes_action.cc]: check mi and miles

**Comment by dgearhart** [comment]
:ship:

**Pull request merged**